lib/tbot/service_database_tunnel.go

```diff
@@ -78,6 +79,19 @@ func (s *DatabaseTunnelService) buildLocalProxyConfig(ctx context.Context) (lpCf
 	ctx, span := tracer.Start(ctx, "DatabaseTunnelService/buildLocalProxyConfig")
 	defer span.End()
 
+	if s.botIdentityReadyCh != nil {
+		select {
+		case <-s.botIdentityReadyCh:
+		default:
+			s.log.InfoContext(ctx, "Waiting for internal bot identity to be renewed before running")
+			select {
+			case <-s.botIdentityReadyCh:
+			case <-ctx.Done():
+				return alpnproxy.LocalProxyConfig{}, nil
```

Comment:
Kind of nitpick territory, but will this empty value be handled optimally downstream? From my quick read I think `alpnproxy.NewLocalProxy()` will probably produce an error (due to `CheckAndSetDefaults()` failing) so we might still log a confusing error instead of exiting quietly.